src/test/java/htsjdk/samtools/cram/RANSInteropTest.java

```diff
@@ -0,0 +1,321 @@
+            // results
+            final ByteBuffer uncompressedBytes = ByteBuffer.wrap(filterEmbeddedNewlines(IOUtils.toByteArray(is)));
+
+            // If Stripe Flag is set, skip the round trip test as encoding is not implemented for this case.
+            if ((params.getFormatFlags() & RANSNx16Params.STRIPE_FLAG_MASK)==0) {
+                final ByteBuffer compressedBytes = ransEncode.compress(uncompressedBytes, params);
+                uncompressedBytes.rewind();
+                Assert.assertEquals(ransDecode.uncompress(compressedBytes), uncompressedBytes);
+            }
+        }
+    }
+
+    @Test (
+            dataProvider = "allRansCodecsAndData",
+            dependsOnMethods = "testGetHTSCodecsCorpus",
+            description = "Compress the original file using htsjdk RANS and compare it with the existing compressed file. " +
+                    "Uncompress the existing compressed file using htsjdk RANS and compare it with the original file.")
+    public void testRANSPreCompressed(
+            final Path uncompressedInteropPath,
+            final RANSEncode<RANSParams> unused,
+            final RANSDecode ransDecode,
+            final RANSParams params,
+            final String compressedInteropDirName) throws IOException, SkipException {
+        if (!RANSInteropTestUtils.isInteropTestDataAvailable()) {
+            throw new SkipException("Interop test data is not available locally");
+        }
+
+        final Path preCompressedInteropPath = getCompressedRANSPath(compressedInteropDirName,uncompressedInteropPath, params);
+
+        try (final InputStream uncompressedInteropStream = Files.newInputStream(uncompressedInteropPath);
+             final InputStream preCompressedInteropStream = Files.newInputStream(preCompressedInteropPath)
+        ) {
+            // preprocess the uncompressed data (to match what the htscodecs-library test harness does)
+            // by filtering out the embedded newlines, and then round trip through RANS and compare the
+            // results
+            final ByteBuffer uncompressedInteropBytes = ByteBuffer.wrap(filterEmbeddedNewlines(IOUtils.toByteArray(uncompressedInteropStream)));
+
+            final ByteBuffer preCompressedInteropBytes = ByteBuffer.wrap(IOUtils.toByteArray(preCompressedInteropStream));
+
+            // Use htsjdk to uncompress the precompressed file from htscodecs repo
+            final ByteBuffer uncompressedHtsjdkBytes = ransDecode.uncompress(preCompressedInteropBytes);
+
+            // Compare the htsjdk uncompressed bytes with the original input file from htscodecs repo
+            Assert.assertEquals(uncompressedHtsjdkBytes, uncompressedInteropBytes);
+        } catch (NoSuchFileException ex){
+            throw new SkipException("Skipping testRANSPrecompressed as either input file " +
+                    "or precompressed file is missing. File Missing: " + ex.getMessage());
```

Comment:
Whenever you're re-throwing a new exception to replace an existing one, its generally best to use the constructor overload for the new exception that takes the existing exception as an arg, rather than extracting the message as you do here. This preserves the call stack for the original exception, which is often very useful since its preserves the root cause of the failure:

`throw new SkipException("Skipping testRANSPrecompressed as either input file or precompressed file is missing.", ex);`

Reply:
done